### PR TITLE
feat(containers): comprehensive metadata extraction with kubernetes integration

### PR DESCRIPTION
**Pull request by jra3**
## Summary

This PR adds comprehensive container metadata extraction capabilities to the Antimetal Agent, enabling enriched observability data for containerized workloads. The implementation extracts container-specific metadata (image info, labels, resource limits, human-readable identifiers) while avoiding duplication of Kubernetes Pod-level data that's already available through K8s resources.

**Key capabilities added:**
- **Image metadata extraction**: Parses container image names, tags, and digests from runtime metadata files
- **Label extraction**: Captures all container labels from Docker/containerd config files
- **Resource limit extraction**: Reads cgroup v1/v2 CPU and memory limits (shares, quota, cpuset, memory limits)
- **Human-readable identifiers**: Generates clean container and workload names with Kubernetes hash stripping
- **Multi-runtime support**: Works seamlessly across Kubernetes, Docker, containerd, CRI-O, Podman, and Docker Compose
- **Graceful degradation**: Returns best-effort metadata when runtime files are missing or inaccessible

## Motivation

Container metadata enrichment is critical for:
1. **Cost attribution**: Linking resource usage to specific workloads and images
2. **Performance analysis**: Understanding resource limits and their impact on container behavior
3. **Security auditing**: Tracking which images and versions are running in production
4. **Workload identification**: Mapping low-level cgroup metrics to logical application names

Without this feature, the agent could only report raw cgroup paths and PIDs, making it difficult for users to understand which applications are consuming resources.

### New Package: `pkg/containers`
- **`metadata.go`**: Core metadata extraction logic (670 lines)
  - `ExtractMetadata()`: Main entry point for metadata extraction
  - Image metadata extraction from multiple runtime-specific paths
  - Label extraction with JSON parsing for Docker/containerd configs
  - Resource limit extraction from cgroup v1/v2 hierarchies
  - Human-readable identifier generation with hash stripping

- **`metadata_test.go`**: Comprehensive test suite (408 lines, 24 test cases)
  - Tests for all major container runtimes (K8s, Docker, Podman, containerd, CRI-O)
  - Edge cases: missing files, malformed JSON, invalid cgroup values
  - Hash stripping validation for Kubernetes workload names

### Integration Points

**`internal/containers/manager.go`** (+48 lines):
- Integrated metadata extraction into container discovery workflow
- Populates `ContainerNode` with extracted metadata fields
- Maintains backwards compatibility with existing discovery logic

**`internal/containers/graph/builder.go`** (+30 lines):
- Updated graph builder to include container metadata in resource nodes
- Added metadata fields to ContainerNode resources for intake streaming

**`internal/containers/graph/nodes.go`** (+14 lines):
- Extended node creation to accept and populate metadata fields

### API Changes

**`pkg/api/antimetal/runtime/v1/linux.pb.go`** (binary protocol buffer update):
- Updated protobuf bindings to match jra3-apis PR #14
- Added fields: `container_name`, `workload_name`, `image_name`, `image_tag`, `labels`, `cpu_shares`, `cpu_quota_us`, `cpu_period_us`, `memory_limit_bytes`, `cpuset_cpus`, `cpuset_mems`

## Dependencies

- **jra3-apis PR #14**: https://github.com/antimetal/apis/pull/14
- Adds `ContainerNode` protobuf schema fields for metadata
- Protobuf bindings included in this PR (generated from jra3-apis changes)

## Testing

**Unit Tests** (24 comprehensive test cases):
- ✅ Image metadata extraction from all runtime paths
- ✅ Label extraction from Docker/containerd configs
- ✅ Resource limit parsing for cgroup v1 and v2
- ✅ Human-readable identifier generation
- ✅ Kubernetes hash stripping (deployment/statefulset/replicaset patterns)
- ✅ Graceful degradation for missing/malformed files
- ✅ Runtime-specific path handling (Docker, K8s, Podman, containerd, CRI-O, Docker Compose)

**Integration Testing**:
- ✅ Tested in KIND cluster with real Kubernetes workloads
- ✅ Validated graceful degradation when metadata files unavailable
- ✅ Confirmed no duplication of Pod-level Kubernetes data

### Hash Stripping Algorithm

Kubernetes appends hash suffixes to workload names (e.g., `web-server-7d4f8b9c5d-abc123`). The implementation strips these hashes to reveal the logical workload name:

```go
// Input: "web-server-7d4f8b9c5d-abc123" (Deployment pod)
// Output: "web-server"

// Input: "nginx-statefulset-0" (StatefulSet pod)
// Output: "nginx-statefulset-0"
```

Supports Deployment, StatefulSet, ReplicaSet, DaemonSet, and Job patterns.

### Runtime-Specific Paths

The implementation searches multiple paths for metadata files, ensuring compatibility across runtimes:

**Image metadata paths**:
- `/sys/fs/cgroup/.../io.kubernetes.cri.image-name` (Kubernetes CRI)
- `/proc/<pid>/root/.dockerenv`, `/proc/<pid>/root/.containerenv` (runtime markers)
- Container config files in `/var/lib/docker`, `/var/lib/containerd`, etc.

**Label paths**:
- `/var/lib/docker/containers/<id>/config.v2.json` (Docker)
- `/var/run/containerd/io.containerd.runtime.v2.task/k8s.io/<id>/config.json` (containerd)
- `/var/lib/containers/storage/overlay-containers/<id>/userdata/config.json` (Podman)

### Resource Limit Extraction

Reads cgroup files with proper v1/v2 detection:

**cgroup v1**:
- `cpu.shares`, `cpu.cfs_quota_us`, `cpu.cfs_period_us`
- `memory.limit_in_bytes`
- `cpuset.cpus`, `cpuset.mems`

**cgroup v2**:
- `cpu.weight` (converted to shares)
- `cpu.max` (quota/period in single file)
- `memory.max`
- `cpuset.cpus`, `cpuset.mems`

## Breaking Changes

None. This PR is additive only:
- Existing container discovery functionality unchanged
- New metadata fields are optional extensions
- Backwards compatible with existing intake service

## Review Checklist

- [x] Code follows project style guidelines (`make fmt`, `make fmt.clang`)
- [x] License headers added (`make gen-license-headers`)
- [x] Unit tests written and passing (24 test cases, 100% coverage)
- [x] Integration tested in KIND cluster
- [x] No breaking changes to existing APIs
- [x] Documentation updated (inline comments, function docs)
- [x] Graceful error handling for missing files
- [x] Multi-runtime compatibility validated

### CODE REVIEW

File: pkg/containers/metadata.go [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+		}
+	}
+	return true
+}
+
+// extractResourceLimits reads cgroup resource limit files and populates the metadata
+func extractResourceLimits(container *Container, metadata *Metadata) {
+	cgroupPath := container.CgroupPath
+
+	if container.CgroupVersion == 1 {
```
[the remark this answers is not in the document]
Comment (jra3) [reply pkg/containers/metadata.go]: I've swapped the order to check v2 first for consistency

It wasn't really needed here because we already did the discovery earlier where we do prefer v2 over v1

**Review by haq204 — APPROVED** [review]
